core/domain/story_domain.py

```diff
@@ -297,16 +301,19 @@ def validate(self):
 class StoryContents(object):
     """Domain object representing the story_contents dict."""
 
-    def __init__(self, story_nodes, starting_node_id):
+    def __init__(self, story_nodes, starting_node_id, next_node_id):
         """Constructs a StoryContents domain object.
 
         Args:
             story_nodes: list(StoryNode). The list of story nodes that are part
                 of this story.
             starting_node_id: str. The id of the starting node of the story.
+            next_node_id: int. The index for the next node to be added to the
```

Comment:
So a node ID is a string. This should be number_for_next_node_id if it's an int, or it should be a string if it's actually meant to be a node ID. (Probably the latter and you can perhaps have a method for increment_next_node_id that does all the manipulation.)

Reply:
Done, changed next_node_id to a string.